### PR TITLE
Add support for aggregation operation in connector

### PR DESCRIPTION
**Pull request by DinithHerath**
## Purpose

- Add aggregation functionality to the connector.

## Approach
- Added aggregation operation by passing collection and stages in json array format to the connector.
- Support all the aggregation operations regardless of the mongo-java-driver being used, as long as the mongodb supports that operation.

### CODE REVIEW

File: src/main/resources/functions/component.xml [hunk 1]
```diff
@@ -50,5 +50,9 @@
             <file>updateMany.xml</file>
             <description>Modifies several documents in a collection.</description>
         </component>
+        <component name="aggregation">
```
Comment (Tharsanan1) [inline src/main/resources/functions/component.xml]: Please make the operation name to 'aggregate'. the operation name should be a verb like insert, find etc

**Review by Tharsanan1 — APPROVED** [review]

File: src/main/resources/functions/aggregate.xml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+<!--
+
+ Copyright (c) 2017, WSO2 LLC (http://www.wso2.org) All Rights Reserved.
```
Comment (dushaniw) [inline src/main/resources/functions/aggregate.xml:3]: change licenser header year and remove `All Rights Reserved.`

File: pom.xml [hunk 3]
```diff
@@ -53,7 +53,7 @@
         <dependency>
             <groupId>org.mongodb</groupId>
             <artifactId>mongo-java-driver</artifactId>
-            <version>3.12.7</version>
+            <version>3.12.12</version>
```
Comment (dushaniw) [inline pom.xml:56]: lets use a property defined for the version and reuse it.

File: src/main/java/org/wso2/carbon/connector/utils/SimpleMongoClient.java [hunk 4]
```diff
@@ -268,6 +270,29 @@ public JSONObject deleteManyDocuments(String collectionName, Document filter, St
         return deleteResult;
     }
 
+    public JSONArray aggregationPipeline(String collectionName, List<Document> pipelines) throws JSONException {
```
Comment (dushaniw) [inline src/main/java/org/wso2/carbon/connector/utils/SimpleMongoClient.java:273]: lets add a method level comment.

File: src/main/java/org/wso2/carbon/connector/operations/Aggregate.java [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+            JSONArray aggregationResult = simpleMongoClient.aggregationPipeline(collection, stages);
+
+            if (log.isDebugEnabled()) {
+                log.debug(AGGREGATION_RESULT + aggregationResult.toString());
+            }
+            MongoUtils.setPayload(messageContext, aggregationResult.toString());
+
+        } catch (BsonInvalidOperationException e) {
+            MongoUtils.handleError(messageContext, e, MongoConstants.MONGODB_CONNECTIVITY, INVALID_DOCUMENTS_MESSAGE);
+
```
Comment (dushaniw) [inline src/main/java/org/wso2/carbon/connector/operations/Aggregate.java:110]: remove the additional empty lines

**Pull request merged**